src/main/java/ro/msg/learning/shop/mapper/OrderMapper.java

```diff
@@ -0,0 +1,39 @@
+package ro.msg.learning.shop.mapper;
+
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+import org.mapstruct.factory.Mappers;
+import ro.msg.learning.shop.dto.OrderDTO;
+import ro.msg.learning.shop.entity.OrderDetailEntity;
+import ro.msg.learning.shop.entity.OrderEntity;
+
+@Mapper
+public interface OrderMapper
+{
+    OrderMapper INSTANCE = Mappers.getMapper(OrderMapper.class);
+
+    @Mapping(source = "oE.orderD.orderDId.product.id", target = "prodIDs")
```

Comment:
Property "oE.orderD.orderDId.product.id" can't be find in OrderEntity. Clearest way to do this is to have multiple sources - for example OrderDTO toOrderDTO(OrderEntity orderEntity, ProductEntity productEntity)